internal/service/streamprivatelinkendpoint/model.go

```diff
@@ -25,14 +26,14 @@ func NewTFModel(ctx context.Context, projectID string, apiResp *admin.StreamsPri
 		ServiceEndpointId:   types.StringPointerValue(apiResp.ServiceEndpointId),
 		State:               types.StringPointerValue(apiResp.State),
 		Vendor:              types.StringPointerValue(apiResp.Vendor),
+		Arn:                 types.StringPointerValue(apiResp.Arn),
 	}
-	if apiResp.DnsSubDomain != nil {
-		subdomain, diags := types.ListValueFrom(ctx, types.StringType, apiResp.GetDnsSubDomain())
-		if diags.HasError() {
-			return nil, diags
-		}
-		result.DnsSubDomain = subdomain
+
+	subdomain, diags := types.ListValueFrom(ctx, types.StringType, apiResp.GetDnsSubDomain())
+	if diags.HasError() {
+		return nil, diags
 	}
+	result.DnsSubDomain = subdomain
```

Comment:
These lines and the related "DnsSubDomain" test changes are from https://github.com/mongodb/terraform-provider-mongodbatlas/pull/3161. I'll rebase them away once the other PR is merged.